Sources/FrontEnd/TypeChecking/TypeChecker.swift

```diff
@@ -3646,9 +3627,41 @@ public struct TypeChecker {
     return r
   }
 
+  /// Returns the declarations in `decls` whose names match `n`.
+  ///
+  /// - Requires: The base name of all declarations in `decls` is equal to `n-stem`
+  mutating func filter(decls: DeclSet, matching n: Name) -> [AnyDeclID] {
```

Comment:
```suggestion
mutating func filter(_ decls: DeclSet, named n: Name) -> [AnyDeclID] {
```

should this be a method of `DeclSet`?  If `DeclSet` is (or has) a `Collection`, you could use the existing `filter` with a suitable predicate.

Reply:
I think it would make the API worse because we need `self` mutating.